llama_index/program/guidance_program.py

```diff
@@ -27,21 +27,40 @@ def __init__(
         verbose: bool = False,
     ):
         try:
-            from guidance import Program
-            from guidance.llms import OpenAI
+            from guidance.models import OpenAI
         except ImportError as e:
             raise ImportError(
                 "guidance package not found." "please run `pip install guidance`"
             ) from e
 
-        llm = guidance_llm or OpenAI("text-davinci-003")
-        output_str = pydantic_to_guidance_output_template_markdown(output_cls)
-        full_str = prompt_template_str + "\n" + output_str
+        if not guidance_llm:
+            llm = guidance_llm
+        else:
+            llm = OpenAI("text-davinci-003")
```

Comment:
No, it will raise errors if we use the gpt-3.5 model (even though they mentioned the gpt-3.5 turbo in the documentation but it doesn't support it).

Reply:
I think it's getting removed like.... today LOL https://platform.openai.com/docs/deprecations

Reply:
Oh, thanks for sharing this, I'll check to find any newer model that supports guidance and will ping you by tagging here after

Reply:
@amindadgar Looks like it might be as simple as `llm = guidance.models.OpenAIChat("gpt-3.5-turbo")`

Reply:
When trying the `llm = guidance.models.OpenAIChat("gpt-3.5-turbo")` I would get the following error for generating the required text (in our case the text is the json array of sub-questions)

```
ValueError: The OpenAI model gpt-3.5-turbo is a Chat-based model and requires role tags in the prompt!             Make sure you are using guidance context managers like `with system():`, `with user():` and `with assistant():`             to appropriately format your guidance program for this type of model.
```

We can solve it by adding the `system():` and `user():` tags in the prompt but then it wouldn't be a general prompt that could be used for any different LLM. Unless you don't have any other suggestions I'll continue working on this to find a suitable LLM in the next day.

Reply:
That's such a strange restriction 😭 we can probably figure out how to add those tags when using guidance though? 

For an llm, it just expects a string prompt template. We can add the user tag to the template string before sending it to the llm? Not sure

Reply:
Yes, we can update the `DEFAULT_SUB_QUESTION_PROMPT_TMPL` in `llama_index.question.gen.prompts` but in case of choosing another llm for the guidance_llm within the `GuidanceQuestionGenerator` of the llama_index library, the performance of it can hugely decrease (maybe someone wanted to use their own llm)

We have two paths ahead of use
1. To find and choose another llm of OpenAI
2. To update the query we're making from the variable `DEFAULT_SUB_QUESTION_PROMPT_TMPL` to include the `system():` and `user():` tags, and add some comments for everyone to update the template when they want to change the default guidance_llm prompt to something else (the prompt can be adjusted in `GuidanceQuestionGenerator.from_defaults`)

I'll try to find a solution with item 1 and if they didn't work or any other issues raised, I'll go with the 2nd item. :)

How does this sound to you?

Reply:
Thanks a ton for the effort! Great news!